test/distributed/test_c10d_common.py

```diff
@@ -66,8 +66,13 @@ def gpus_for_rank(world_size):
     On a single node, all visible GPUs are evenly
     divided to subsets, each process only uses a subset.
     """
-    visible_devices = list(range(torch.cuda.device_count()))
-    gpus_per_process = torch.cuda.device_count() // world_size
+    device_count = (
+        torch.xpu.device_count()
+        if torch.xpu.is_available()
+        else torch.cuda.device_count()
+    )
```

Comment:
It would be good to move it to some `torch/testing/_internal`  as say `get_device_count()` or something like that? Because you are making similar change in 3-4 places in the code

Reply:
Make sense. We will do some code generalize even for tests.